src/gluonts/model/deepar/_schema.py

```diff
@@ -0,0 +1,161 @@
+# Copyright 2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License").
+# You may not use this file except in compliance with the License.
+# A copy of the License is located at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# or in the "license" file accompanying this file. This file is distributed
+# on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+# express or implied. See the License for the specific language governing
+# permissions and limitations under the License.
+
+from dataclasses import Field
+from typing import Any, Dict, Optional, Union, Generic, TypeVar, Type
+
+import numpy as np
+import pandas as pd
+from functools import lru_cache
+from pandas.tseries.frequencies import to_offset
+from pandas.tseries.offsets import Tick
+from gluonts.exceptions import GluonTSDataError
+
+
+T = TypeVar("T")
+
+
+class FieldType(Generic[T]):
+    def __call__(self, v: Any) -> T:
+        raise NotImplementedError()
+
+    def __eq__(self, other):
+        raise NotImplementedError()
+
+    def is_compatible(self, v: Any) -> bool:
+        raise NotImplementedError()
+
+
+class NumpyArrayField(FieldType[np.ndarray]):
+    def __init__(
+        self,
+        dtype: Type = np.float32,
+        ndim: Optional[int] = None,
+    ) -> None:
+        self.dtype = dtype
+        self.ndim = ndim
+
+    def __eq__(self, other):
+        if not isinstance(other, NumpyArrayField):
+            return False
+        return self.dtype == other.dtype and self.ndim == other.ndim
+
+    def __repr__(self):
+        return f"NumpyArrayField(dtype={self.dtype!r}, ndim={self.ndim!r}"
+
+    def __call__(self, value: Any) -> np.ndarray:
+        value = np.asarray(value, dtype=object)
+        value = value.astype(self.dtype)
```

Comment:
Why do this in two steps?

Reply:
By doing so, we can convert string type to our target type. `asarray` can’t handle it.

Reply:
Can you provide an example?

Reply:
I think this doesn't work too. What I what to solve is converting float string data like '1.2' to int data 1.
```
>>> a = np.asarray(['1.2', '2.3'], dtype=float)
>>> print(a)
[1.2 2.3]
>>> a = np.asarray(['1.2', '2.3'], dtype=int)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
ValueError: invalid literal for int() with base 10: '1.2'
>>> a = np.asarray(['1.2', '2.3'], dtype=object)
>>> print(a.astype(int))
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
ValueError: invalid literal for int() with base 10: '1.2'
```